extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -656,21 +697,25 @@ export class AmpImageSlider extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    // TODO(kqian): remove after launch
     // From https://github.com/ampproject/amphtml/pull/16688
     user().assert(isExperimentOn(this.win, 'amp-image-slider'),
         'Experiment <amp-image-slider> disabled');
 
-    // TODO(kqian): remove this after layer launch
-    if (!isExperimentOn(this.win, 'layers')) {
-      this.scheduleLayout(dev().assertElement(this.leftAmpImage_));
-      this.scheduleLayout(dev().assertElement(this.rightAmpImage_));
-    }
+    // Extensions such as amp-carousel still uses .setAsOwner()
+    // This would break the rendering of the images as carousel
+    // will call .scheduleLayout on the slider but not the contents
+    // while Resources would found amp-imgs' parent has owner and
+    // refuse to run the normal scheduling in discoverWork_.
+    // SIMPLER SOL: simply always call scheduleLayout no matter what
+    this.scheduleLayout(dev().assertElement(this.leftAmpImage_));
```

Comment:
It seems like this logic needs to be consistent with the `setAsOwner` logic in `buildCallback`. I don't mind which logic you use so long as it's well-commented (this one here is pretty good, you can just move the comment up to buildCallback and reference the comment here).

Reply:
I think it would be better have the comment still here, since the section in buildCallback is mostly address the layers experiment. I already have a pointer from buildCallback that points to this comment